### PR TITLE
[QF-921]: introduces country_lanaguage_preference endpoint

### PR DESCRIPTION
**Pull request by mohsinayoob**
closes #[QF-921](https://quranfoundation.atlassian.net/browse/QF-921)

[QF-921]: https://quranfoundation.atlassian.net/browse/QF-921?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/controllers/api/qdc/resources_controller.rb [hunk 1]
```diff
@@ -116,6 +116,49 @@ def languages
       render
     end
 
+    def country_language_preference
+      user_device_language = request.query_parameters[:user_device_language]
+      country = request.query_parameters[:country]
+    
+      if user_device_language.blank?
+        return render_bad_request('user_device_language is required')
+      end
```
Comment (osamasayed) [inline app/controllers/api/qdc/resources_controller.rb:125]: Don't we need to also validate the language is valid e.g. someone enters `abc`

File: app/controllers/api/qdc/resources_controller.rb [hunk 2]
```diff
@@ -116,6 +116,49 @@ def languages
...
+    
+      if country.blank?
```
Comment (osamasayed) [inline app/controllers/api/qdc/resources_controller.rb:127]: Don't we need to also validate the country is valid e.g. someone enters `abc`

File: app/controllers/api/qdc/resources_controller.rb [hunk 3]
```diff
@@ -116,6 +116,49 @@ def languages
...
+          learning_plan_languages: @learning_plan_languages
+        }
+        render
+      else
+        render_404("No matching country language preference found")
+      end
+    end
+    
+    def load_default_resources
+      @default_mushaf = @preference.default_mushaf_id ? Mushaf.find(@preference.default_mushaf_id) : nil
```
Comment (osamasayed) [inline app/controllers/api/qdc/resources_controller.rb:153]: We should filter by enabled Musahfs only

File: app/controllers/api/qdc/resources_controller.rb [hunk 4]
```diff
@@ -116,6 +116,49 @@ def languages
...
+      @default_translations = @preference.default_translation_ids.present? ? ResourceContent.where(id: @preference.default_translation_ids.split(',')) : []
+      @default_tafsir = @preference.default_tafsir_id ? ResourceContent.find(@preference.default_tafsir_id) : nil
```
Comment (osamasayed) [inline app/controllers/api/qdc/resources_controller.rb:155]: We should filter tafsirs and translations by approved only

**Review by osamasayed — CHANGES_REQUESTED** [review]

File: db/migrate/20240918072240_create_country_language_preferences.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      t.string :default_wbw_language
+      t.integer :default_reciter
+      t.string :ayah_reflections_languages
+      t.string :learning_plan_languages
+
+      t.timestamps
+    end
+
+    add_foreign_key :country_language_preferences, :reciters, column: :default_reciter
+  end
```
[the remark this answers is not in the document]
Comment (mohsinayoob) [reply db/migrate/20240918072240_create_country_language_preferences.rb:18]: In most cases we are using languages table in quran schema and also the other which are referring to language, they have relation with languages table of quran schema not i18n schema.
The issue here is we have to add unique constraint to iso_code filed of languages table to use it as foreign key. 
Making that change too

**Review by osamasayed — APPROVED** [review]

**Pull request merged**